dowhy/causal_prediction/algorithms/base_algorithm.py

```diff
@@ -0,0 +1,59 @@
+import torch
+from torch.nn import functional as F
+import pytorch_lightning as pl
+
+class Algorithm(pl.LightningModule):
```

Comment:
needs a docstring for all methods in this class.

docstring should provide: one line summary of the method, then a description of each parameter, and then a description of what the method returns.

Reply:
These are default methods from Pytorch lightning module (https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#training) and are called within the loop when the `fit` function is called. Hence, the parameter descriptions are not very useful I think and nothing is returned as such.

I have now added descriptions though similar to lightning documentation and mentioned that `training_step` needs to be overidden for new algorithms. Also, that the test loop is called only when `test()` is explicitly called.

Reply:
Okay, in that case, can you add a docstring to the class, saying that "This class implements the default methods for a  Pytorch lightning module. Its methods are called when the fit() method is called. To know more about these methods, refer to <link to pytorch lightning docs>"

Reply:
Can we also rename this class? `Algorithm` could refer to a lot of things. `CausalPredictionBaseAlgorithm` may be a better name?